### PR TITLE
introduce global Singleton to managers

### PR DESCRIPTION
**Pull request by volatile-static**

### CODE REVIEW

File: src/managers/itemTree.ts [hunk 1]
```diff
@@ -257,15 +258,10 @@ export class ItemTreeManager extends ManagerTool {
...
-    const window = this.getGlobal("window");
-    if (!Zotero._ItemTreeExtraColumnsGlobal) {
-      let globalCache = {
-        columns: [],
-        fieldHooks: {},
-        renderCellHooks: {},
-      } as ItemTreeExtraColumnsGlobal;
-      this.globalCache = Zotero._ItemTreeExtraColumnsGlobal = globalCache;
-
+    const window = this.getGlobal("window"),
```
Comment (windingwind) [inline src/managers/itemTree.ts:261]: ',' at the end of line

File: src/managers/itemTree.ts [hunk 2]
```diff
@@ -257,15 +258,10 @@ export class ItemTreeManager extends ManagerTool {
...
+      globalCache = this.globalCache = ToolkitGlobal.getInstance(Zotero).itemTree;
+    if (!globalCache.initialized) {
+      globalCache.initialized = true;
```
Comment (windingwind) [inline src/managers/itemTree.ts:264]: Possible confliction between multiple plugins.

Suppose A is just about to start initializing globalCache, B finds globalCache.initialized is true and thus try to access its props/methods, while they are not initialized yet.

Add a wait lock or load start & load finish flags for the initialization.

File: src/managers/toolkitGlobal.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    private constructor() {
+        this.itemTree = {
+            initialized: false,
+            columns: [],
+            fieldHooks: {},
+            renderCellHooks: {},
+        };
+    }
+
+    static getInstance(zotero: _ZoteroTypes.Zotero): ToolkitGlobal {
```
Comment (windingwind) [inline src/managers/toolkitGlobal.ts:15]: Discussion: getInstance here, or getGlobal in basicTool?

File: src/managers/itemTree.ts (hunk 1 above)
Comment (volatile-static) [reply src/managers/itemTree.ts:261]: that is a const declaration

File: src/managers/toolkitGlobal.ts (hunk 3 above)
Comment (volatile-static) [reply src/managers/toolkitGlobal.ts:15]: you can add a wrap for this in `getGlobal`

File: src/managers/itemTree.ts (hunk 2 above)
Comment (volatile-static) [reply src/managers/itemTree.ts]: I'm not sure the ready-idel-loding states is safe

File: src/managers/toolkitGlobal.ts (hunk 3 above)
Comment (volatile-static) [reply src/managers/toolkitGlobal.ts:15]: In addition, `getInstance` here is necessary for Singleton.

**Pull request merged**